localization/autoware_localization_evaluation_scripts/scripts/analyze_rosbags_parallel.py

```diff
@@ -0,0 +1,62 @@
+#!/usr/bin/env python3
+"""A script to analyze rosbags in parallel."""
+
+import argparse
+from multiprocessing import Pool
+from pathlib import Path
+
+import compare_trajectories
+import extract_pose_from_rosbag
+import plot_diagnostics
+
+
+def parse_args() -> argparse.Namespace:
+    parser = argparse.ArgumentParser()
+    parser.add_argument("result_dir", type=Path)
+    parser.add_argument("--parallel_num", type=int, default=1)
+    parser.add_argument("--topic_subject", type=str, default="/localization/kinematic_state")
+    parser.add_argument(
+        "--topic_reference", type=str, default="/localization/pose_estimator/pose_with_covariance"
+    )
+    return parser.parse_args()
+
+
+def process_directory(directory: Path, topic_subject: str, topic_reference: str) -> None:
+    target_rosbag = directory / "result_bag"
+    compare_result_dir = directory / "compare_trajectories"
+    compare_result_dir.mkdir(parents=True, exist_ok=True)
+
+    plot_diagnostics.main(rosbag_path=target_rosbag)
+
+    save_name_subject = extract_pose_from_rosbag.topic_name_to_save_name(topic_subject)
+    save_name_reference = extract_pose_from_rosbag.topic_name_to_save_name(topic_reference)
+
+    extract_pose_from_rosbag.main(
+        rosbag_path=target_rosbag,
+        target_topics=[
+            topic_subject,
+            topic_reference,
+        ],
+        save_dir=compare_result_dir,
+    )
+
+    compare_trajectories.main(
+        subject_tsv=compare_result_dir / f"{save_name_subject}.tsv",
+        reference_tsv=compare_result_dir / f"{save_name_reference}.tsv",
+    )
+
+
+if __name__ == "__main__":
+    args = parse_args()
+    result_dir = args.result_dir
+    parallel_num = args.parallel_num
+    topic_subject = args.topic_subject
+    topic_reference = args.topic_reference
+
+    directories = sorted([d for d in args.result_dir.iterdir() if d.is_dir()])
```

Comment:
The output directory of driving_log_replayer_v2 has the symlink to the latest result.
How about the following to avoid the symlink?


```suggestion
    directories = sorted([d for d in args.result_dir.iterdir() if d.is_dir() and not d.is_symlink()])
```

Reply:
Fix https://github.com/autowarefoundation/autoware_tools/pull/217/commits/40f6e426ed7170a6cbe5f2a13bc765e59d4b2b95

Reply:
Thanks for the update.
The script no longer processes the latest directory twice!

```shell
# before
❯ ros2 run autoware_localization_evaluation_scripts analyze_rosbags_parallel.py /home/hayatomizushima/dlr2_data/localization/out
/usr/lib/python3/dist-packages/scipy/__init__.py:146: UserWarning: A NumPy version >=1.17.3 and <1.25.0 is required for this version of SciPy (detected version 1.26.4
  warnings.warn(f"A NumPy version >={np_minversion} and <{np_maxversion}"
save_dir=PosixPath('/home/hayatomizushima/dlr2_data/localization/out/2025-0227-105308/diagnostics_result')
/localization/kinematic_state: 1389 msgs
/localization/pose_estimator/pose_with_covariance: 274 msgs
Saved pose tsv files to /home/hayatomizushima/dlr2_data/localization/out/2025-0227-105308/compare_trajectories
mean error: 0.056 m
saved to /home/hayatomizushima/dlr2_data/localization/out/2025-0227-105308/compare_trajectories/localization__kinematic_state_result/relative_pose.png
save_dir=PosixPath('/home/hayatomizushima/dlr2_data/localization/out/2025-0227-204542/diagnostics_result')
/localization/kinematic_state: 1285 msgs
/localization/pose_estimator/pose_with_covariance: 253 msgs
Saved pose tsv files to /home/hayatomizushima/dlr2_data/localization/out/2025-0227-204542/compare_trajectories
mean error: 0.055 m
saved to /home/hayatomizushima/dlr2_data/localization/out/2025-0227-204542/compare_trajectories/localization__kinematic_state_result/relative_pose.png
save_dir=PosixPath('/home/hayatomizushima/dlr2_data/localization/out/latest/diagnostics_result')
/localization/kinematic_state: 1285 msgs
/localization/pose_estimator/pose_with_covariance: 253 msgs
Saved pose tsv files to /home/hayatomizushima/dlr2_data/localization/out/latest/compare_trajectories
mean error: 0.055 m
saved to /home/hayatomizushima/dlr2_data/localization/out/latest/compare_trajectories/localization__kinematic_state_result/relative_pose.png


# after
❯ ros2 run autoware_localization_evaluation_scripts analyze_rosbags_parallel.py /home/hayatomizushima/dlr2_data/localization/out
/usr/lib/python3/dist-packages/scipy/__init__.py:146: UserWarning: A NumPy version >=1.17.3 and <1.25.0 is required for this version of SciPy (detected version 1.26.4
  warnings.warn(f"A NumPy version >={np_minversion} and <{np_maxversion}"
save_dir=PosixPath('/home/hayatomizushima/dlr2_data/localization/out/2025-0227-105308/diagnostics_result')
/localization/kinematic_state: 1389 msgs
/localization/pose_estimator/pose_with_covariance: 274 msgs
Saved pose tsv files to /home/hayatomizushima/dlr2_data/localization/out/2025-0227-105308/compare_trajectories
mean error: 0.056 m
saved to /home/hayatomizushima/dlr2_data/localization/out/2025-0227-105308/compare_trajectories/localization__kinematic_state_result/relative_pose.png
save_dir=PosixPath('/home/hayatomizushima/dlr2_data/localization/out/2025-0227-204542/diagnostics_result')
/localization/kinematic_state: 1285 msgs
/localization/pose_estimator/pose_with_covariance: 253 msgs
Saved pose tsv files to /home/hayatomizushima/dlr2_data/localization/out/2025-0227-204542/compare_trajectories
mean error: 0.055 m
saved to /home/hayatomizushima/dlr2_data/localization/out/2025-0227-204542/compare_trajectories/localization__kinematic_state_result/relative_pose.png
```